### DISCUSSION TITLE
Is there an & missing in the Language tour; Basic types; Buffers, arrays, and slices array example?

### DISCUSSION THREAD

**Discussion by marvin-the-mathematician** [hylo-lang/hylo-lang.github.io#67] [Announcements]
Was working my way through the Language tour and got to the section (https://tour.val-lang.dev/basic-types#buffers-arrays-and-slices) and in the sub-section named Buffers, arrays, and slices I noticed no & on the line marked HERE below:
```
typealias Point = (x: Double, y: Double)
public fun main() {
  var points = Array<Point>()
  print(points.count())            // 0
  points.append((x: 6.3, y: 1.0))  // <== HERE
  print(points.count())            // 1
}
```

Should it be:
```
  ...
  &points.append((x: 6.3, y: 1.0))  // <== HERE
  ...
```
?
I just want to make sure I understand the use of & :-) Thanks.

**Comment by kyouko-taiga** [comment]
Good catch, there's indeed a missing mutation marker in the example.

**Reply by kyouko-taiga** [reply]
The error has been corrected. Thanks for reporting it!

**Reply by marvin-the-mathematician** [reply]
Cool. Thanks for the super quick fix (it looks good to me from this end now). Apologies for the pedantry; I just thought that since this is one of the core features of the language (explicit mutation markers) not having it there in the tutorial might lead to a bunch of unnecessary confusion :-)

In that spirit, I also expected to see them later on in the section on Structures:
```
type Matrix3 {
  public var components: Double[3][3]
  public memberwise init
}

public fun main() {
  var m = Matrix3(components: [
    [0 ,0, 0],
    [0 ,0, 0],
    [0 ,0, 0],
  ])
  m.components[0][0] = 1.0
  m.components[1][1] = 1.0
  m.components[2][2] = 1.0
  print(m)
}
```
, i.e. like this:
```
type Matrix3 {
  public var components: Double[3][3]
  public memberwise init
}

public fun main() {
  var m = Matrix3(components: [
    [0 ,0, 0],
    [0 ,0, 0],
    [0 ,0, 0],
  ])
  &m.components[0][0] = 1.0
  &m.components[1][1] = 1.0
  &m.components[2][2] = 1.0
  print(m)
}
```
But perhaps they are not needed there for some reason?

**Reply by kyouko-taiga** [reply]
They should be needed. We wrote the tutorial before we took the decision to require mutation markers for assignments.

Again, good catch, thanks!

**Reply by marvin-the-mathematician** [reply]
Cool. The preceding docs must have taught me enough to spot these small typos anyway; so that's a good sign :-)

P.S. I just learned about Val through attending Sean Parent's talk "All the Safeties" at Cpp-On-Sea 2023. I have followed Sean's talks for a long time now and am a big fan after having adopted a whole host of his Better Code advice over the years, so I'm exited to see how all that good stuff shows up in the Val language :-)
